lib/model/EetplanModel.class.php

```diff
@@ -1,141 +1,75 @@
 <?php
 
+require_once 'model/entity/Eetplan.class.php';
+require_once 'model/entity/EetplanBekenden.class.php';
+
 /**
  * EetplanModel.class.php
  * 
  * @author C.S.R. Delft <pubcie@csrdelft.nl>
+ * @author G.J.W. Oolbekkink <g.j.w.oolbekkink@gmail.com>
  * 
  * Verzorgt het opvragen van eetplangegevens
  */
-class EetplanModel {
+class EetplanModel extends PersistenceModel {
+
+    const orm = 'Eetplan';
+
+    private $lichting;
+
+    /**
+     * EetplanModel constructor.
+     * @param string $lichting Lichting om eetplan voor op te halen, 2 cijfers.
+     */
+    public function __construct($lichting)
+    {
+        parent::__construct();
+        $this->lichting = $lichting;
+    }
 
-	function getDatum($iAvond) {
-		$aAvonden = array(
-			'28-10-2014',
-			'25-11-2014',
-			'20-01-2015',
-			'17-02-2015',
-			'12-05-2015',
-			'09-06-2015'
-		);
-		if ($iAvond > 0 AND $iAvond <= sizeof($aAvonden)) {
-			return $aAvonden[$iAvond - 1];
-		}
-	}
+    /**
+     * Haal alle avonden op die voor deze lichting gelden.
+     *
+     * @return Eetplan[] Lijst met sparse(!) eetplan objecten met alleen een avond.
+     */
+    public function getAvonden() {
+        return $this->findSparse(array('avond'), 'uid LIKE ?', array("$this->lichting%"), 'avond')->fetchAll();
```

Comment:
Heb je hier fetchAll() nodig? Als je foreach gebruikt hoeft dat namelijk niet.

FYI Doc zegt verder over findSparse: (is nu zo te zien niet van toepassing)
>Get the (non-sparse) attributes and their values of this object.
***Relies on getters and setters to update $attr_retrieved***